src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -223,6 +223,10 @@ private static bool IsTupleExpression(BoundKind kind)
             {
                 var field = fields[i];
 
+                // Use default field rather than implicitly named fields since
+                // fields from inferred names are not usable in C# 7.0.
+                field = field.CorrespondingTupleField ?? field;
```

Comment:
`CorrespondingTupleField` is not `null` for (non-default) named fields for the first 7 elements. It is `null` for default fields for the first 7 elements and for any fields after the 7th element.

The use-site diagnostic check will be hit for inferred names after the 7th element and for any use-site errors on the underlying field type, regardless of whether the field is named. I'll investigate adding tests for those two cases.